server/dav/backend/directory.js

```diff
@@ -154,13 +162,24 @@ var jsDAV_Chunked_Directory = module.exports = jsDAV_FS_Directory.extend(jsDAV_F
           handler.dispatchEvent("afterBind", handler.httpRequest.url,
                                 Path.join(self.path, filename));
 
-          Etag(originalPath, cbfswritechunk);
+          self.getETag(cbfswritechunk);
         });
       } else {
         cbfswritechunk(null, null);
       }
     });
 
     handler.getRequestBody(type, stream, false, function() {});
+  },
+
+  // We redefine fsext's `delete` because it uses asyncjs rmtree, which causes
+  // stack overflows on very large directories.
+  "delete": function(cbfsdel) {
+    var self = this;
+    ChildProcess.spawn("rm", ["-rf", this.path]).on("exit", function(err) {
```

Comment:
It _should_ be protected by https://github.com/mnutt/davros/blob/master/server/dav/backend/tree.js#L23 but I just filed this earlier today: https://github.com/mikedeboer/jsDAV/issues/149